### ISSUE TITLE
electron-updater error TS6053: File 'typings/electron.d.ts' not found

### ISSUE THREAD

**Issue by Noobsai** [electron-userland/electron-builder#8264]
* **Electron-Builder Version**:  24.13.3
    "electron-updater": "6.2.1",

* **Node Version**: v22.3.0

* **Electron Version**: N/A
* **Electron Type (current, beta, nightly)**: N/A

* **Target**: N/A

Cannot build project after upgrading npm package `electron-updater` from `6.1.8` to `6.2.1` with "skipLibCheck": false in TS-config:
```
node_modules/electron-updater/out/AppUpdater.d.ts:1:22 - error TS6053: File 'D:/project-name/typings/electron.d.ts' not found.

1 /// <reference path="../../../typings/electron.d.ts" />
                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
node_modules/electron-updater/out/electronHttpExecutor.d.ts:1:22 - error TS6053: File 'D:/project-name/typings/electron.d.ts' not found.

1 /// <reference path="../../../typings/electron.d.ts" />
                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Found 2 error(s).
```
It seems the npm package has been broken since version `6.1.9`.

Workaround:
- set `skipLibCheck` = true in tsconfig.json
- downgrade to `6.1.8`

**Comment by srizzon** [comment]
Same issue here, but didn't work when downgrading to `6.1.8`. Any solution?

**Comment by matiasnl23** [comment]
I'm encountering the same error using the latest Electron and electron-updater. I don't know if I'm doing something wrong, but it looks for the typing/electron.d.ts file in my root folder instead of looking for it in node_modules or another appropriate location.

**Comment by mmaietta** [comment]
So I took a look at this and ran into a blocker. The electron.d.ts file is a copy-paste from electron and not supposed to be bundled with electron-updater, but is necessary to avoid compiler errors for things such as `process.resourcesPath`.
https://github.com/electron-userland/electron-builder/blob/2b80b01c4e9e5e7693460adda789cf270fbefe2b/typings/electron.d.ts#L1-L4

When resyncing my electron-builder changes with the typings file moved within electron-updater, my test projects renders hundreds of errors like this one below, likely due to the fact that electron type definitions are automatically installed as part of `electron` npm package and I'm assuming these two type definition files are conflicting/overlapping.
```
node_modules/electron/electron.d.ts:7602:13 - error TS2320: Interface 'SystemPreferences' cannot simultaneously extend types 'EventEmitter' and 'EventEmitter'.
  Named property 'prependListener' of types 'EventEmitter' and 'EventEmitter' are not identical.

7602   interface SystemPreferences extends NodeJS.EventEmitter {
```

Not sure how to resolve this issue to let all projects be able to use most recent type definitions corresponding to their installed electron version.

**Comment by Irelander** [comment]
same issue, and I managed to work around it by setting skipLibCheck=true. It would be great if we could find a proper solution rather than just a workaround!

**Comment by mmaietta** [comment]
Should be fixed in [electron-updater@6.3.1](https://github.com/electron-userland/electron-builder/releases/tag/electron-updater%406.3.1)

Note: As with all electron-updater version updates/upgrades, please be very diligent in your runtime testing.
